### PR TITLE
Add check release PR action

### PR DESCRIPTION
**Pull request by rlh1994**
## Description

This PR adds a new action to automatically check the following on any PR that goes from a branch starting with `release` to the `main` branch:
- Last commit is called `prepare for release`
- Changelog first line is in the correct format and has a date of today or in the future
- Version in project yaml matches changelog
- Version in integration test project yaml matches the changelog
- The newest version is semver higher than the previous one

Technically there is nothing that checks there has been ANY update, but the date one will fail unless it was done the same day as another release?

As it's an action can move straight to the main branch no release needed.

You can see an (eventually) working example in this pr https://github.com/snowplow/dbt-snowplow-unified/pull/65

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] ✅ Test
- [ ] 🤖 Build
- [x] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

https://snplow.atlassian.net/browse/PE-6301

## Checklist

- [ ] 💣 Is your change a breaking change?
- [ ] 📖 I have updated the CHANGELOG.md

### Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📓 internal package docs (ymls, macros, readme, if applicable)
- [ ] 📕 I have raised a [Snowplow documentation](https://github.com/snowplow/documentation) PR if applicable (Link here)
- [x] 🙅 no documentation needed

### CODE REVIEW

File: .github/scripts/check_release.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    if len(old_versions) < 2:
+        logs.append(
+            "⚠️ Warning: Not enough versions found in changelog to perform semver comparison"
+        )
+        return
+
+    next_newest_version = old_versions[1]
+    if semver.compare(new_version, next_newest_version) <= 0:
+        logs.append(
+            f"❌ Error: New version (`{new_version}`) is not greater than the next newest version (`{next_newest_version}`)"
```
Comment (agnessnowplow) [inline .github/scripts/check_release.py:89]: you mean latest (active)?

File: CHANGELOG [hunk 2]
```diff
@@ -1,3 +1,7 @@
+snowplow-unified 0.4.4 (2024-06-25)
+---------------------------------------
```
Comment (agnessnowplow) [inline CHANGELOG:2]: Why is it needed here? Thought we can test this upon opening a new pr against main AFTER this is live?

File: .github/scripts/check_release.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
+import os
+import re
+import sys
+from datetime import datetime
+import yaml
+import semver
+
+logs = []
+
+
```
Comment (agnessnowplow) [inline .github/scripts/check_release.py:10]: Thanks for this! A few less things to worry about double checking! 🥳

File: .github/workflows/release-pr-checks.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
+name: Release PR Checks
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  check-release:
+    if: startsWith(github.head_ref, 'release')
```
[the remark this answers is not in the document]
Comment (rlh1994) [reply .github/workflows/release-pr-checks.yml:10]: I think so yes

File: CHANGELOG (hunk 2 above)
Comment (rlh1994) [reply CHANGELOG:2]: Accident, shouldn't be here :D

**Review by agnessnowplow — APPROVED** [review]
LGTM

**Pull request merged**